src/ParseUser.js

```diff
@@ -869,7 +869,7 @@ const DefaultController = {
   updateUserOnDisk(user) {
     const path = Storage.generatePath(CURRENT_USER_KEY);
     const json = user.toJSON();
-    json.className = '_User';
+    json.className = user.className || '_User';
```

Comment:
how about 

`json.className = this.constructor.name === 'ParseUser' ? '_User' : this.constructor.name;`

i think that'll work and then you don't have to worry about setting this.className.

Reply:
Perfect!

Reply:
Looks like we have a flaky test. I can't restart the build the button is missing, can you?